pandas/_libs/tslibs/np_datetime.pyx

```diff
@@ -624,9 +664,12 @@ cdef int64_t convert_reso(
     else:
         # e.g. ns -> us, risk of overflow, but no risk of lossy rounding
         mult = get_conversion_factor(from_reso, to_reso)
-        with cython.overflowcheck(True):
```

Comment:
is there a cython bug?

Reply:
I don't think it's a bug per se. I think Cython let's the overflow happen but then adds checks after the fact to see if it overflowed. This by contrast prevents the overflow from happening in the first place. It generally gets you to the same place in the end

Reply:
for my edification, this pattern is considered Better Practice than the one cython uses?

Reply:
I think the only difference here is that this will make the sanitizer happy whereas the cython approach will not